src/toolkit/res_manip.cc

```diff
@@ -65,6 +68,34 @@ std::vector<Resource::Ptr> ResCast(std::vector<Resource::Ptr> rs) {
   return rs;
 }
 
+double GetFillMass(Resource::Ptr r, Package::Ptr pkg) {
+  if (r->quantity() == 0) {
+    throw Error("cannot fill zero resources into a package");
+  }
+  else if (r->quantity() < pkg->fill_min()) {
+    // less than one pkg of material available
+    return 0;
+  }
```

Comment:
I'm guessing we need to treat these two cases differently because we can have `fill_min == 0`.

thought: If we force `fill_min` to always bigger than `eps` than we can treat `quantity == 0` as the same as `quantity < fill_min`?

I'm not sure if that helps/hurts us down the line with material accountancy?

Alternatively, we can just return 0 in both cases and not throw an error?